### ISSUE TITLE
improving scheduler performance and reliability

### ISSUE THREAD

**Issue by kares** [elastic/logstash#13773]
This issue is intended to track `Rufus::Scheduler` updates due production issues.

For now, issues exhibit with multiple schedulers (~ 20) started from multiple pipelines e.g. having JDBC input.
There might be multiple bugs involved, the list of changes since rufus-scheduler 3.0.9 is long.

- [x] https://github.com/logstash-plugins/logstash-integration-jdbc/pull/102 (helps debugging the issue on production)
- [ ] https://github.com/logstash-plugins/logstash-integration-jdbc/pull/103

Eventually the changes (in JDBC input) should be shared across all plugins that use the scheduler:
(for now this is on hold due https://github.com/elastic/logstash/issues/12931 - unblocking plugins to be able to use latest scheduler)
- [ ] test and make sure all plugins work with latest rufus-scheduler (but continue working with 3.0.9 as well) #12931
- [ ] create a scheduler mixin with some utility method to start (and stop) a scheduler instance
- [ ] consider an upstream patch that completely avoids `Thread.list`
